python/sparta.py

```diff
@@ -56,33 +56,36 @@ def close(self):
     self.spa = None
 
   def file(self,file):
-    self.lib.sparta_file(self.spa,file)
+    b_file = file.encode('utf-8')
+    self.lib.sparta_file(self.spa,c_char_p(b_file))
 
   def command(self,cmd):
-    self.lib.sparta_command(self.spa,cmd)
+    b_cmd = cmd.encode('utf-8')
+    self.lib.sparta_command(self.spa,c_char_p(b_cmd))
 
   def extract_global(self,name,type):
+    b_name = name.encode('utf-8')
```

Comment:
That is what the u''-String prefix does in your Python program. The main question is whether we want to let the user do that in the scripts or do that in an interface layer.

Reply:
ah, I see what your earlier comment meant now.  I don't think we should assume or require the user to pass in a byte-string.